lsp-mode.el

```diff
@@ -5662,7 +5663,7 @@ the symbol to rename at point."
   :group 'lsp-mode
   :type 'boolean)
 
-(defun lsp--get-symbol-to-rename ()
+(cl-defun lsp--get-symbol-to-rename ()
```

Comment:
Why convert this to `cl-defun`? You're not adding named parameter here

Reply:
Because I use `cl-return`, and that needs an implicit `nil` block. I could also wrap this in `cl-block`, but that would cause extra-indentation.

Reply:
Good catch, it didn't do what I thought.

Reply:
The `cl-block` for `cl-defun` has the name of the function while `cl-return` is returning from block `nil`.

Reply:
I just realized this too, which is why I made this PR draft.

Reply:
The control flow here will require further thought.